src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/aws/AwsCli.kt

```diff
@@ -21,21 +28,47 @@ class AwsCli {
         val lock = LOCKS.computeIfAbsent(ssh.getHost().ipAddress) { Object() }
         synchronized(lock) {
             val awsCliExecutionResult = ssh.safeExecute("aws --version", Duration.ofSeconds(30), Level.TRACE, Level.TRACE)
-            if (!awsCliExecutionResult.isSuccessful()) {
+            if (awsCliExecutionResult.isSuccessful()) {
+                val combinedOutput = "${awsCliExecutionResult.output}${awsCliExecutionResult.errorOutput}"
+                require(combinedOutput.contains("aws-cli/$cliVersion")) {
+                    "Aws Cli version $cliVersion requested but different version is already installed: '${combinedOutput}'."
+                }
+            } else {
                 Ubuntu().install(ssh, listOf("zip", "python"), Duration.ofMinutes(3))
-                ssh.execute(
-                    cmd = "curl --silent https://s3.amazonaws.com/aws-cli/awscli-bundle-1.15.51.zip -o awscli-bundle.zip",
-                    timeout = Duration.ofSeconds(50)
-                )
-                ssh.execute("unzip -n -q awscli-bundle.zip")
-                ssh.execute(
-                    cmd = "sudo ./awscli-bundle/install -i /usr/local/aws -b /usr/local/bin/aws",
-                    timeout = Duration.ofSeconds(60)
-                )
+                val majorVersion = versionRegex.find(cliVersion)?.groupValues?.get(1)
+                when (majorVersion) {
+                    "1" -> installV1Cli(ssh)
+                    else -> installV2Cli(ssh)
+                }
             }
         }
     }
 
+    private fun installV1Cli(ssh: SshConnection) {
+        ssh.execute(
+            cmd = "curl --silent https://s3.amazonaws.com/aws-cli/awscli-bundle-$cliVersion.zip -o awscli-bundle.zip",
+            timeout = Duration.ofSeconds(50)
+        )
+        ssh.execute("unzip -n -q awscli-bundle.zip")
+        ssh.execute(
+            cmd = "sudo ./awscli-bundle/install -i /usr/local/aws -b /usr/local/bin/aws",
+            timeout = Duration.ofSeconds(60)
+        )
+    }
+
+    private fun installV2Cli(ssh: SshConnection) {
+        // Instructions from https://docs.aws.amazon.com/cli/latest/userguide/getting-started-version.html
```

Comment:
Nit: Wouldn't it be better as a method kdoc? It would be more clear that the comment is about the whole method and not only line 61